apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/StandardResponseAdapters.kt

```diff
@@ -125,8 +128,47 @@ object UploadResponseAdapter : ResponseAdapter<Upload> {
   }
 }
 
-fun <T: Any> ResponseAdapter<T>.nullable() = NullableResponseAdapter(this)
+class ObjectResponseAdapter<T>(
+    private val wrappedAdapter: ResponseAdapter<T>,
+    private val buffered: Boolean
+) : ResponseAdapter<T> {
+  override fun fromResponse(reader: JsonReader, responseAdapterCache: ResponseAdapterCache): T {
+    val actualReader = if (buffered) {
+      reader.buffer()
+    } else {
+      reader
+    }
+    actualReader.beginObject()
+    return wrappedAdapter.fromResponse(actualReader, responseAdapterCache).also {
+      actualReader.endObject()
+    }
+  }
+
+  override fun toResponse(writer: JsonWriter, responseAdapterCache: ResponseAdapterCache, value: T) {
+    if (buffered && writer !is MapJsonWriter) {
+      /**
+       * Convert to a Map first
+       */
+      val mapWriter = MapJsonWriter()
+      mapWriter.beginObject()
+      wrappedAdapter.toResponse(mapWriter, responseAdapterCache, value)
+      mapWriter.endObject()
+
+      /**
+       * And write to the original writer
+       */
+      AnyResponseAdapter.toResponse(writer, mapWriter.root())
+    } else {
+      writer.beginObject()
+      wrappedAdapter.toResponse(writer, responseAdapterCache, value)
+      writer.endObject()
+    }
+  }
+}
+
+fun <T : Any> ResponseAdapter<T>.nullable() = NullableResponseAdapter(this)
 fun <T> ResponseAdapter<T>.list() = ListResponseAdapter(this)
+fun <T> ResponseAdapter<T>.obj(buffered: Boolean = false) = ObjectResponseAdapter(this, buffered)
```

Comment:
should we use more readable name like `.toObjectResponseAdapter` or `.objectResponseAdapter`, `asObjectResponseAdapter`

Reply:
I'd like a "short" name as this ends up being used all over the place in generated adapters:

```kotlin
Friends.obj().nullable().list().fromResponse(reader, responseAdapterCache)

// Compared to
Friends.asObjectResponseAdapter().asNullableResponseAdapter().asListResponseAdapter().fromResponse(reader, responseAdapterCache)
```

Since there can be a bunch of chained calls, keeping them short will make sure it fits on one line in most cases and it's displayed nicely.  Ideally, I was aiming for `object()` but this is a hard keyword in kotlin so that doesn't work. Any other suggestion of short names?